int32 allied_type
float32 hue
---
int32 error